cmd/snap/cmd_run.go

```diff
@@ -1107,22 +1136,28 @@ func (x *cmdRun) runCmdUnderStrace(origCmd []string, envForExec envForExecFunc)
 	return err
 }
 
-func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook string, beforeExec func() error, args []string) error {
+func (x *cmdRun) runSnapConfine(info *snap.Info, component *snap.ComponentInfo, securityTag string, snapTarget string, hook *snap.HookInfo, beforeExec func() error, args []string) error {
 	snapConfine, err := snapdHelperPath("snap-confine")
 	if err != nil {
 		return err
 	}
 	if !osutil.FileExists(snapConfine) {
-		if hook != "" {
-			logger.Noticef("WARNING: skipping running hook %q of snap %q: missing snap-confine", hook, info.InstanceName())
+		if hook != nil {
+			logger.Noticef("WARNING: skipping running hook %q of snap %q: missing snap-confine", hook.Name, info.InstanceName())
 			return nil
 		}
 		return fmt.Errorf(i18n.G("missing snap-confine: try updating your core/snapd package"))
 	}
 
 	logger.Debugf("executing snap-confine from %s", snapConfine)
 
-	snapName, appName := snap.SplitSnapApp(snapApp)
+	var snapName string
+	if hook != nil {
+		snapName, _ = snap.SplitSnapComponentInstanceName(snapTarget)
+	} else {
+		snapName, _ = snap.SplitSnapApp(snapTarget)
+	}
```

Comment:
I don't understand the if here ? do we need to make the difference if  snap.SplitSnapComponentInstanceName works in all cases?

Reply:
I think this was handled better by the refactor in the following commit.